knimin/handlers/ag_pulldown.py

```diff
@@ -41,8 +41,10 @@ def post(self):
         metadata, failures = db.pulldown(barcodes, blanks)
 
         meta_zip = InMemoryZip()
-        failtext = ("The following barcodes were not retrieved for any "
-                    "survey:\n%s" % "\n".join(failures))
+        failed = '\n'.join(['\t'.join(x) for x in
+                            db.explain_pulldown_failures(failures)])
```

Comment:
I suppose it could, but the pulldown code relies on the list returned by pulldown to check if the pulldown works for a specific barcode or not. I could check by length of failures rather than actual value.